cdc/kv/matcher.go

```diff
@@ -62,12 +69,15 @@ func (m *matcher) putPrewriteRow(row *cdcpb.Event_Row) {
 func (m *matcher) matchRow(row *cdcpb.Event_Row, initialized bool) bool {
 	if value, exist := m.unmatchedValue[newMatchKey(row)]; exist {
 		// TiKV may send a fake prewrite event with empty value caused by txn heartbeat.
-		//
 		// We need to skip match if the region is not initialized,
 		// as prewrite events may be sent out of order.
 		if !initialized && len(value.GetValue()) == 0 {
 			return false
 		}
+		// Pipelined-DML transactions can only be matched after initialized.
+		if !initialized && value.Generation > 0 {
```

Comment:
Is it possible that the `Generation` is 0 ?

Reply:
Yes. Normal transactions it will be 0. Only in pipelined-dml transacitons it will be greater than 0.